fix: correct reference to MIN_EPOCHS for data columns

dguenther opened the pull request:
**Motivation**

Found a couple more places we should be referencing MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS.

**Description**

* Updated `migrateDataColumnSidecarsFromHotToColdDb ` to reference `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS`
* Updated `chain.archiveDataEpochs` to reference `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS`

dguenther committed:
fix: correct reference to MIN_EPOCHS for data columns

nflaig commented on packages/cli/src/options/beaconNodeOptions/chain.ts:
```diff
@@ -277,7 +277,8 @@ Will double processing times. Use only for debugging purposes.",
   "chain.archiveDataEpochs": {
-    description: "Number of epochs to retain finalized blobs (minimum of MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS)",
+    description:
+      "Number of epochs to retain finalized blobs (minimum of MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS/MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS)",
```
in fulu we aren't retaining full blobs anymore, just columns so the flag description still seems a bit off to me

maybe

```suggestion
      "Number of epochs to retain finalized blobs/columns (minimum of MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS/MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS)",
```

dguenther committed:
Update packages/cli/src/options/beaconNodeOptions/chain.ts

Co-authored-by: Nico Flaig <nflaig@protonmail.com>

dguenther commented:
Good suggestion, updated!

nflaig commented:
[failed unit test](https://github.com/ChainSafe/lodestar/actions/runs/14803769851/job/41568764387?pr=7788#step:5:2870) seems related

dguenther committed:
fix: Fix incorrect config in data column archive test

dguenther commented:
Updated, I'm fairly sure the unit test was correct but needs to be updated for this change. Previously:

* `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS` = 1
* current epoch = 2
* blocks created for slots 1-6, epoch 0

So the blocks would be filtered out of `migrateDataColumnSidecarsFromHotToColdDb`. Updated `MIN_EPOCHS_FOR_DATA_COLUMN_SIDECARS_REQUESTS` to 2, since the unit test isn't intending to test expired data columns

nflaig reviewed (approved):
seems good to me now, thanks @dguenther

wemeetagain commented:
:tada: This PR is included in [v1.34.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.34.0) :tada:
